scripts/generate_integration_test.py

```diff
@@ -0,0 +1,133 @@
+    post_data = parse_qs(request.post_data)
+    del post_data["csrf_token"]
+
+    items: dict[str, str | list[str]] = {}
+    for answer_id, answer_values in post_data.items():
+        if len(answer_values) > 1:  # Handle multiple values in list
+            items[answer_id] = answer_values
+        elif len(answer_values) == 1:
+            items[answer_id] = answer_values[0]  # A single item in list
+
+    if items:
+        file.write(generate_method_request("post", items))
+    else:
+        # Empty post for no answers and non question pages
+        file.write(generate_method_request("post", ""))
+
+
+# pylint: disable=global-statement
+def process_get(request: Request, file: TextIO) -> None:
+    global previous_request_method
+    global survey_start
+
+    if (  # Logic to filter out the session and root questionnaire GET requests at the start
+        previous_request_method == "GET"
+        and "session?token" not in request.url
+        and request.url != f"{RUNNER_ROOT_URL}/questionnaire/"
+    ):
+        path = f'"{urlparse(request.url).path}"'
+        file.write(generate_method_request("get", path))
+
+    if not survey_start:
+        previous_request_method = request.method
+
+    if (
+        previous_request_method == ""
+        and survey_start
+        and request.url == f"{RUNNER_ROOT_URL}/questionnaire/"
+    ):
+        survey_start = False
+
+
+def process_launcher_request(request: Request) -> None:
+    global schema_name
+
+    if request.method == "GET":
+        if survey_start:
+            # start of journey, so create a skeleton file using the schema name
+            schema_name = parse_qs(request.url)["schema_name"][0]
+            with open(f"./scripts/{schema_name}.py", "w", encoding="utf-8") as file:
+                file.write(
+                    f"from tests.integration.integration_test_case import IntegrationTestCase\n\n\n"
+                    f'class {schema_name.title().replace("_", "")}(IntegrationTestCase):\n'
+                    f"    def {schema_name}(self):\n"
+                    f'        self.launchSurvey("{schema_name}")'
+                )
+                logger.info(f"self.launchSurvey('{schema_name}')")
+        else:
+            # capture launcher urls for sign-out, save etc
+            with open(f"./scripts/{schema_name}.py", "a", encoding="utf-8") as file:
+                path = f'"{urlparse(request.url).path}"'
+                file.write(generate_method_request("get", path))
+
+
+def generate_method_request(method: str, data: dict | str | None = "") -> str:
+    snippet = f"self.{method}({data})"
+    logger.info(snippet)
+
+    return f"\n        {snippet}"
+
+
+def request_handler(request: Request) -> None:
+    if LAUNCHER_ROOT_URL in request.url:
+        process_launcher_request(request)
+    elif RUNNER_ROOT_URL in request.url:
+        process_runner_request(request)
+
+
+def run(pw: Playwright) -> None:
```

Comment:
We tried updating the ini file but couldn't get it working because when mypy [is invoked](https://github.com/ONSdigital/eq-questionnaire-runner/blob/main/scripts/run_mypy.sh#L3) it has the `app` folder. We added the script name in there instead as we couldn't find a graceful way around it?